File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    print(goal)
    answer = ''
    cnt = 0
    print(len(goal))
    
    for i in range(0,len(goal)):
        print(i)
        if cards1 and goal[i] == cards1[0]:
            cards1.remove(goal[i])
            
        elif cards2 and goal[i] == cards2[0] :
            cards2.remove(goal[i])

        else:
            answer = "No"
            break
        cnt += 1
        
    if cnt == len(goal):
        answer = "Yes"
    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    
    let cnt1 = cards1.length
    let cnt2 = cards2.length
    let result = goal.length
    
    goal.forEach((value) => {
        if(cnt1 !== 0 && value === cards1[cards1.length-cnt1]) {
            cnt1--
            result--
        } else if (cnt2 !== 0 && value === cards2[cards2.length-cnt2]) {
            cnt2--
            result--
        }
    })
    
    if(result === 0) {
        return "Yes"
    }else {
        return "No"
    }
}